tf2_geometry_msgs/scripts/test.py

```diff
@@ -159,18 +159,22 @@ def test_transform(self):
         v.pose.pose.orientation = Quaternion(w=0.0, x=1.0, y=0.0, z=0.0)
 
         out = tf2_geometry_msgs.do_transform_pose_with_covariance_stamped(v, t)
-        expected_covariance = (
+        expected_covariance = np.array([
           1.0, -2.0, -3.0, 4.0, -5.0, -6.0,
           -1.0, 2.0, 3.0, -4.0, 5.0, 6.0,
           -1.0, 2.0, 3.0, -4.0, 5.0, 6.0,
           1.0, -2.0, -3.0, 4.0, -5.0, -6.0,
           -1.0, 2.0, 3.0, -4.0, 5.0, 6.0,
           -1.0, 2.0, 3.0, -4.0, 5.0, 6.0
-        )
+        ])
         self.assertEqual(out.pose.pose.position.x, 2)
         self.assertEqual(out.pose.pose.position.y, 0)
         self.assertEqual(out.pose.pose.position.z, 0)
-        self.assertEqual(out.pose.covariance, expected_covariance)
+        self.assertEqual(out.pose.orientation.x, 0)
+        self.assertEqual(out.pose.orientation.y, 0)
+        self.assertEqual(out.pose.orientation.z, 0)
+        self.assertEqual(out.pose.orientation.w, 1)
+        self.assertTrue(np.array_equiv(out.pose.covariance, expected_covariance))
```

Comment:
In hindsight, the purpose of the comparison here may fit [`np.array_equal`](https://numpy.org/doc/stable/reference/generated/numpy.array_equal.html) (check for shape and element equality) over [`np.array_equiv`](https://numpy.org/doc/stable/reference/generated/numpy.array_equiv.html) (check for shape broadcast-ability and element equality) - since we want to avoid the output pose covariance from being ill-formed during the transformation step, i.e. we should check that the `do_transform` function does not return an output covariance that is not in the `(X,)` shape.

Thoughts?

Reply:
Also in hindsight, I was just wondering whether I should've mentioned it in my previous comment, but I thought since you made the suggestion, you would've been aware of it. But reading the documentation of both, it definitely seems like `np.array_equal` is the choice to go by, as I cannot think of a situation where we would need to broadcast some other array shape into this shape since the covariance will always be `(36,)`. So I will make this change.

And also something I thought of mentioning earlier is that the `@` operator for matrix multiplication is only available from Python 3.5 onwards (which has already been EOL'd) and the alternative for numpy matrix multiplication is `A.dot(B)`.

Reply:
> is only available from Python 3.5 onwards

I believe that's fine because of what you said, and also because the default Python version in current OS's tends to be higher than that.